### ISSUE TITLE
Finagle support for tracing and metrics

### ISSUE THREAD

**Issue by greggdonovan** [census-instrumentation/opencensus-specs#134]
Finagle is a popular JVM RPC system [used at](https://github.com/twitter/finagle/blob/develop/ADOPTERS.md) Twitter, FourSquare, Tumblr, Etsy, etc. 

Finagle has existing [tracing support for Zipkin](https://github.com/twitter/finagle/tree/develop/finagle-zipkin-core) and exposes a [large amount of metrics](https://twitter.github.io/finagle/guide/Metrics.html).

**Comment by Sebruck** [comment]
Could be either part of Finagle itself or part of https://github.com/Sebruck/opencensus-scala , we already have instrumentations for akka-http and http4s there.

**Comment by adriancole** [comment]
historically speaking, it is likely best to get some of the core finaglers
in on this as they'll have more stake in how things go. zipkin for example,
it actually not zipkin as much as finagle's internal tracing library. For
example, finagle's tracing library is not used by anything but finagle, and
it isn't directly linked to zipkin either.

Probably you could make a project like zipkin-finagle
<https://github.com/openzipkin/zipkin-finagle> which uses census instead as
a start as it is a lot less work than ripping all of the internal
infrastructure out for something else. Finagle is managed as a monorepo, so
basically it is a fairly big effort to track instrumentation directly...
doing a finagle tracer impl is a lot lower hanging fruit.

**Comment by ramonza** [comment]
Should this issue be moved to the opencensus-java?

**Comment by songy23** [comment]
This is for both Scala and Java, I think it makes more sense to keep it in this repo.

**Comment by LiuVII** [comment]
Hey guys!
Wonder if there's an update on this?

**Comment by pjanotti** [comment]
@LiuVII not sure if it is an option in your case, but one possibility is to use the OC Agent/Collector receiving zipkin and exporting in the format that you need.

**Comment by adriancole** [comment]
or trace exporter even. I don't think agent would be required. This would
work for the few options census has but there are no direct zipkin
reporters for.
